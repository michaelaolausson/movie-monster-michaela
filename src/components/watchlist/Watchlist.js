import React, { useState, useEffect } from "react";
import { Container, Row, Col, Alert } from "react-bootstrap";
import Movie from "./../movie/Movie";
import TMDB from "../../api/tmdb/TMDB";
const tmdb = new TMDB();

const Watchlist = () => {
  const [error, setError] = useState(null);
  const [movies, setMovies] = useState([]);

  useEffect(() => {
    if (localStorage.session_id) {
      getWatchlist();
    } else {
      window.location.href = "/auth";
    }
  }, []);

  const getWatchlist = async () => {
    try {
      const response = await tmdb.getWatchlist();
      setMovies(response.data.results);
      setError(null);
    } catch (exception) {
      setError("Couldn't get watchlist :(");
    }
  };

  if (movies && movies.length > 0) {
    return (
      <Container>
        <Row>
          <Col>
            <h2>
              Watchlist{" "}
              <span role="img" className="float-right" aria-label="clock-emoji">
                🕒
              </span>
            </h2>
          </Col>
        </Row>
        <Row>
          {movies.map((movie, index) => (
            <Movie
              watchlist={true}
              id={movie.id}
              key={index}
              title={movie.title}
              year={movie.release_date}
              description={movie.overview}
              imagePath={movie.poster_path}
            />
          ))}
        </Row>
      </Container>
    );
  } else if (error) {
    return (
      <Container>
        <Row>
          <Col>
            <Alert id="error-box" variant="danger">{error}</Alert>
          </Col>
        </Row>
      </Container>
    );
  }
  return (
    <Container>
      <Row>
        <Col>
          <h2>
            Watchlist
            <span role="img" className="float-right" aria-label="clock-emoji">
              🕒
            </span>
          </h2>
        </Col>
      </Row>
      <Row>
        <Col>
          <p className="text-muted">
            It's empty here!
            <span role="img" aria-label="sad-emoji">
              😥
            </span>
          </p>
          <p className="text-muted">
            Find movies to add to your watchlist using our great
            <a href="/search">search</a>
            <span role="img" aria-label="heart-eyes-emoji">
              😍
            </span>
          </p>
        </Col>
      </Row>
    </Container>
  );
};

export default Watchlist;
