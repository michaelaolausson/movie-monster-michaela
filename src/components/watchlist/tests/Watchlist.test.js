import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { shallow, mount, render, unmount } from "enzyme";
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Watchlist from "./../Watchlist";
Enzyme.configure({ adapter: new Adapter() });



describe("<Watchlist />", () => { 
    const wrapper = render(<Watchlist />);

    it("should match the snapshot", () => {
        expect(wrapper.html()).toMatchSnapshot();
      });
})