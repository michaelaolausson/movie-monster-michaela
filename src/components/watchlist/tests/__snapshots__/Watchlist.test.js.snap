// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Watchlist /> should match the snapshot 1`] = `"<div class=\\"row\\"><div class=\\"col\\"><h2>Watchlist<span role=\\"img\\" class=\\"float-right\\" aria-label=\\"clock-emoji\\">&#x1F552;</span></h2></div></div><div class=\\"row\\"><div class=\\"col\\"><p class=\\"text-muted\\">It&apos;s empty here!<span role=\\"img\\" aria-label=\\"sad-emoji\\">&#x1F625;</span></p><p class=\\"text-muted\\">Find movies to add to your watchlist using our great<a href=\\"/search\\">search</a><span role=\\"img\\" aria-label=\\"heart-eyes-emoji\\">&#x1F60D;</span></p></div></div>"`;
